tests/test_parameters_accumulate_gradient_in_fp32.py

```diff
@@ -343,12 +343,12 @@ def _test_tied_weights_sync_with_grad_accum_in_fp32(
                 (
                     target,
                     (
-                        parallel_context.world_rank_matrix[
-                            dist.get_rank(parallel_context.expert_pg),
-                            get_pp_rank_of(target, module=mdl),
-                            dist.get_rank(parallel_context.dp_pg),
-                            dist.get_rank(parallel_context.tp_pg),
-                        ],
+                        parallel_context.get_global_rank(
```

Comment:
There are two more occurences of `world_rank_matrix` in `src/nanotron/trainer.py` can you modify those as well?

Reply:
Hello, thank you very much for your response! I have committed these changes, now I will see if I can create a unit test for it.

Reply:
@0xkerem you could use this as a reference: https://github.com/huggingface/nanotron/blob/d059fcae1d58da30bb2d6672fb9d8982dc0368be/tests/test_distributed.py#L14-L42

Reply:
@xrsrke I've been trying to solve an error since yesterday but I couldn't succeed. This error is not caused by nanotron, I don't want to be misunderstood. It's something from CUDA. It started after an update but I haven't figured it out yet. I removed all the Python environment and libraries and reinstalled them to solve the errors caused by the versions. At first, `torch.cuda.is_available()` was returning false. I fixed this but for now I still have problems with `flash_attn`. I'll try to figure this out tomorrow. If I can't, I commit the test code, but I cannot run the test myself because all tests are ignored (probably due to flash_attn).

Reply:
@0xkerem i just endabled CI/CD for your PR, so now you could run the unit test in our 8x gpus

Reply:
Hey! I try to click on each section on github and Google a lot before asking because I don't want to bother you. I couldn't find where I could use this 8x GPUs for my PR. I thought I would be tested automatically after the commit, so I made a small test commit to try this, but in the `checks` tab it says This workflow requires approval from a maintainer. Sorry for your time.

Reply:
@xrsrke https://github.com/huggingface/nanotron/actions

Reply:
@0xkerem https://github.com/huggingface/nanotron/actions

Reply:
ah right, i just approved your cicd